src/logic/Quotient.java

```diff
@@ -406,187 +51,142 @@ public Set<Channel> getOutputs() {
 
 
-                for (Move move : movesFromRight) {
-                    System.out.println("Rule 6");
+                for (Move move : s_moves) {
                     Move newMoveRule6 = new Move(location, inc, new ArrayList<>());
                     newMoveRule6.setGuards(move.getEnabledPart().conjunction(negated));
                     newMoveRule6.setUpdates(new ArrayList<>(Collections.singletonList(new ClockUpdate(newClock, 0))));
                     resultMoves.add(newMoveRule6);
                 }
             }
 
-            // rule 3+4
-            if (right.getOutputs().contains(channel)) {
-                List<Move> movesFromRight = right.getNextMoves(locRight, channel);
-                // take all moves from right in order to gather the guards and negate them
-                CDD CDDFromMovesFromRight = CDD.cddFalse();
-                for (Move move : movesFromRight) {
-                    CDDFromMovesFromRight = CDDFromMovesFromRight.disjunction(move.getEnabledPart());
-                }
-                CDDFromMovesFromRight = CDDFromMovesFromRight.negation().removeNegative();
-
-                System.out.println("Rule 3/4");
-                Move newMove4 = new Move(location, univ, new ArrayList<>());
-                newMove4.setGuards(CDDFromMovesFromRight);
-                resultMoves.add(newMove4);
+            // rule 7
+            if (Objects.equals(a.getName(), this.newChan.getName())) {
+                Log.debug("Rule 7");
+                Move newMoveRule7 = new Move(location, inc, new ArrayList<>());
+                // invariant is negation of invariant of left conjuncted with invariant of right
+                CDD negatedInvar = lt.getInvariantCddNew().negation();
+                CDD combined = negatedInvar.conjunction(ls.getInvariantCddNew());
+
+                newMoveRule7.setGuards(combined);
+                newMoveRule7.setUpdates(new ArrayList<>(Collections.singletonList(new ClockUpdate(newClock, 0))));
+                resultMoves.add(newMoveRule7);
             }
 
-            // rule 5
-            if (!right.getActions().contains(channel)) {
-                List<Move> movesFrom1 = left.getNextMoves(locLeft, channel);
-
-                for (Move move : movesFrom1) {
-                    System.out.println("Rule 5");
-                    SymbolicLocation newLoc = new ComplexLocation(new ArrayList<>(Arrays.asList(move.getTarget(), locRight)));
-                    ((ComplexLocation) newLoc).removeInvariants();
-                    Move newMove3 = new Move(location, newLoc, new ArrayList<>());
-                    CDD targetInvar = move.getTarget().getInvariant();
-                    targetInvar = targetInvar.transitionBack(move);
-                    newMove3.setGuards(move.getGuardCDD().conjunction(targetInvar));
-                    newMove3.setUpdates(move.getUpdates());
-                    resultMoves.add(newMove3);
+            // rule 8
```

Comment:
```suggestion
            // Rule 8
```

Reply:
There is a new rule 8, so the old rules 8, 9, and 10 are now 9, 10, and 11, respectively. See https://discord.com/channels/812704528175464458/836933691769880616/1004652728636674149 for the paper with this change.